submissions/zalevskyi/friends-app/app.js

```diff
@@ -0,0 +1,133 @@
+}
+function sort(userData, parameters) {
+  if (parameters.hasOwnProperty('sort')) {
+    let sortedData = sortAscending(userData, parameters.sort)
+    if (parameters.hasOwnProperty('order') && parameters['order']==='descending') sortedData.reverse()
+    return sortedData
+  }
+  else return userData
+}
+function sortAscending(userData, propertyToSort) {
+  if (propertyToSort === 'name') return sortAscendingFullName(userData, 'name', 'surname')
+  else if (propertyToSort === 'surname') return sortAscendingFullName(userData, 'surname', 'name')
+  else if (propertyToSort === 'age') return sortAscendingNumeric(userData, 'age')
+  else return sortAscendingString(userData, propertyToSort)
+}
+function sortAscendingFullName(userData, firstToSort, secondToSort) {
+  let sortedData = userData.slice()
+  sortedData.sort((a,b) => {return `${a[firstToSort]} ${a[secondToSort]}`.localeCompare(`${b[firstToSort]} ${b[secondToSort]}`)})
+  return sortedData
+}
+function sortAscendingNumeric(userData, propertyToSort) {
+  let sortedData = userData.slice()
+  sortedData.sort((a,b) => {return a[propertyToSort] - b[propertyToSort]})
+  return sortedData
+}
+function sortAscendingString(userData, propertyToSort) {
+  let sortedData = userData.slice()
+  sortedData.sort((a,b) => {
+    if (a[propertyToSort] > b[propertyToSort]) return 1
+    else if (a[propertyToSort] < b[propertyToSort]) return -1
+    else return 0})
+  return sortedData
+}
+function filter(userData, parameters) {
+  Object.keys(parameters).forEach(key => {
+    if (['name', 'surname', 'country', 'city'].includes(key)) {
+      userData = userData.filter(user => user[key].toLowerCase().startsWith(parameters[key]))}
+    else if (key==='gender') {
+      userData = userData.filter(user => user.gender.toLowerCase()===parameters.gender)}
+    else if (key==='age_from') {
+      userData = userData.filter(user => user.age >= parameters.age_from)}
+    else if (key==='age_till') {
+      userData = userData.filter(user => user.age <= parameters.age_till)}
+  })
+  return userData
+}
+function getUserCard(userData) {
+  return [`<div class='card'>`,
+            `<img class='photo' alt='${userData.name} ${userData.surname} photo' src='${userData.image128px}'>`,
+            `<p class='name'>${userData.name} ${userData.surname}</p>`,
+            `<p class='demographics card_text'>${userData.gender}, ${userData.age} years</p>`,
+            `<p class='city card_text'>${userData.city}</p>`,
+            `<p class='country card_text'>${userData.country}</p>`,
+            `<p class='phone card_text'>${userData.phone}</p>`,
+          `</div>`].join('')
+}
+function getSortLinks(currentSort) {
+  return ['name', 'surname', 'age', 'gender', 'city', 'country'].map(property => {
+    if (property === currentSort) return `<span class='sorting_selected'>${property}</span>`
+    else return `<a href='#sort=${property}' class='sorting_option'>${property}</a>`
```

Comment:
Why it should be link?

Reply:
redone with select (for property to sort) and radio (for order)